common/src/com/sony/transmitpower/observer/TelephonyStateObserver.java

```diff
@@ -108,8 +106,15 @@ public static boolean isDataActive(@NonNull final Context context)
 
     // for each SIM:
     private final class PhoneStateListenerImpl extends PhoneStateListener {
-        PhoneStateListenerImpl(int subId) {
-            super(subId);
+        // Keep track of the used manager for deregistration purposes.
+        private final TelephonyManager mTelephonyManager;
+
+        PhoneStateListenerImpl(TelephonyManager manager) {
```

Comment:
super();
While the default constructor gets inserted automatically, changes in PhoneStateListener constructors may add unnecessary issues. Always better to be explicit.

Reply:
Good one!

(See, Java is not my forte...)

Reply:
Though, now that think about it some more, I don't see an obvious reason why.

If the parameterless constructor disappears, this will fail compilation _anyway_. Do you have an example where this prevents said "unnecessary issues"?

Reply:
I'd rather the compilation fails for an obvious reason, hence the explicit call.

Reply:
Allright, let's do it.

I checked this in a random android app project, and the error is the same for both cases (invalid number of arguments). However, the inline error squiglies make it more obvious. Without `super()` it reports "There is no default constructor available in Foo", and with `super()` it reports `Foo(int) in Foo cannot be applied to ()`. The latter is more vague in my opinion, as it sounds more like a lambda function that is applied to a tuple of arguments.
But whatever, that doesn't matter in this case :grin: